engine/userguide/storagedriver/device-mapper-driver.md

```diff
@@ -504,196 +466,303 @@ The `Data Space` values show that the pool is 100GB total. This example extends
+    Logical volume docker/thinpool_tdata successfully resized.
+    ```
 
-    d.  Reload the thin pool with the new sector number
+4.  Verify the new thin pool size using the `Data Space Available` field in the
+    output of `docker info`. If you extnded the `thinpool/metadata` logical
+    volume instead, look for `Metadata Space Available`.
 
     ```bash
-    $ sudo dmsetup suspend docker-8:1-123141-pool \
-    && sudo dmsetup reload docker-8:1-123141-pool --table '0 419430400 thin-pool 7:1 7:0 128 32768 1 skip_block_zeroing' \
-    && sudo dmsetup resume docker-8:1-123141-pool
+    Storage Driver: devicemapper
+     Pool Name: docker-thinpool
+     Pool Blocksize: 524.3 kB
+     Base Device Size: 10.74 GB
+     Backing Filesystem: xfs
+     Data file:
+     Metadata file:
+     Data Space Used: 212.3 MB
+     Data Space Total: 212.6 GB
+     Data Space Available: 212.4 GB
+     Metadata Space Used: 286.7 kB
+     Metadata Space Total: 1.07 GB
+     Metadata Space Available: 1.069 GB
+    <output truncated>
     ```
 
-#### The device_tool
+## How the `devicemapper` storage driver works
 
-The Docker's projects `contrib` directory contains not part of the core
-distribution. These tools that are often useful but can also be out-of-date. <a
-href="https://goo.gl/wNfDTi">In this directory, is the `device_tool.go`</a>
-which you can also resize the loop-lvm thin pool.
+> **Warning**: Do not directly manipulate any files or directories within
+> `/var/lib/docker/`. These files and directories are managed by Docker.
 
-To use the tool, compile it first. Then, do the following to resize the pool:
+Use the `lsblk` command to see the devices and their pools, from the operating
+system's point of view:
 
 ```bash
-$ ./device_tool resize 200GB
-```
-
-### For a direct-lvm mode configuration
-
-In this example, you extend the capacity of a running device that uses the
-`direct-lvm` configuration. This example assumes you are using the `/dev/sdh1`
-disk partition.
-
-1.  Extend the volume group (VG) `vg-docker`.
+$ sudo lsblk
 
-    ```bash
-    $ sudo vgextend vg-docker /dev/sdh1
+NAME                    MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
+xvda                    202:0    0    8G  0 disk
+└─xvda1                 202:1    0    8G  0 part /
+xvdf                    202:80   0  100G  0 disk
+├─docker-thinpool_tmeta 253:0    0 1020M  0 lvm  
+│ └─docker-thinpool     253:2    0   95G  0 lvm  
+└─docker-thinpool_tdata 253:1    0   95G  0 lvm  
+  └─docker-thinpool     253:2    0   95G  0 lvm
+```
 
-    Volume group "vg-docker" successfully extended
-    ```
+Use the `mount` command to see the mount-point Docker is using:
 
-    Your volume group may use a different name.
+```bash
+$ mount |grep devicemapper
+/dev/xvda1 on /var/lib/docker/devicemapper type xfs (rw,relatime,seclabel,attr2,inode64,noquota)
+```
 
-2.  Extend the `data` logical volume (LV) `vg-docker/data`
+When you use `devicemapper`, Docker stores image and layer contents in
```

Comment:
Images and layer contents are stored in the thinpool, and exposed by mounting them under subdirectories of `/var/lib/docker/devicemapper/`.